spring-boot-task/src/main/java/com/example/task5/repository/DBDataStore.java

```diff
@@ -0,0 +1,246 @@
+
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            preparedStatement.close();
+            return createListTournament(resultSet);
+        }
+    }
+
+    @SneakyThrows
+    public List<Object> getTournament(Integer id) {
+        List<Object> objects = new ArrayList<>();
+
+        PreparedStatement preparedStatement0 = connection
+                .prepareStatement(
+                        "SELECT * FROM TOURNAMENT WHERE ID = " + id);
+        try (ResultSet resultSet = preparedStatement0.executeQuery()) {
+            objects.add(createListTournament(resultSet));
+        }
+
+        String query = "SELECT " +
+                "t1.ID, " +
+                "t1.NAME, " +
+                "t1.CAPTAIN, " +
+                "t1.COACH, " +
+                "t2.ID, " +
+                "t2.NAME, " +
+                "t2.CAPTAIN, " +
+                "t2.COACH, " +
+                "GAMES.ROUND, " +
+                "GAMES.RESULT " +
+                "FROM GAMES " +
+                "LEFT JOIN TEAMS t1 ON GAMES.FIRST_TEAM = t1.ID " +
+                "LEFT JOIN TEAMS t2 ON GAMES.SECOND_TEAM = t2.ID " +
+                "WHERE GAMES.TOURNAMENT = ?";
+
+        PreparedStatement preparedStatement = connection.prepareStatement(query);
+        preparedStatement.setInt(1, id);
+        List<Game> games = new ArrayList<>();
+        try (ResultSet rs = preparedStatement.executeQuery()) {
+            while (rs.next()) {
+                Team team1 = new Team(
+                        rs.getString(2),
+                        rs.getString(3),
+                        rs.getString(4));
+                team1.setId(rs.getInt(1));
+
+                Team team2 = new Team(
+                        rs.getString(6),
+                        rs.getString(7),
+                        rs.getString(8));
+                team2.setId(rs.getInt(5));
+                games.add(new Game(
+                        team1,
+                        team2,
+                        rs.getString(9),
+                        rs.getString(10)));
+            }
+        }
+        objects.add(games);
+        return objects;
+    }
+
+    @SneakyThrows
+    @Override
+    public List<Object> createTournament(String name, List<Team> teams) {
+        PreparedStatement preparedStatement = connection
+                .prepareStatement(
+                        "INSERT INTO TOURNAMENT (NAME) VALUES (?)",
+                        Statement.RETURN_GENERATED_KEYS);
+        preparedStatement.setString(1, name);
+        preparedStatement.executeUpdate();
+
+        int j;
+        try (ResultSet resultSet = preparedStatement.getGeneratedKeys()) {
+            resultSet.next();
+            j = resultSet.getInt(1);
+        }
+        PreparedStatement preparedStatement1 = connection
+                .prepareStatement(
+                        "INSERT INTO TEAMS (NAME, CAPTAIN, COACH) VALUES(?, ?, ?)",
```

Comment:
To improve readability you should create atomic operations: `createTeam`, `createGame`, and so on...